src/scenes/MembershipMintPage/MerkleTree.ts

```diff
@@ -0,0 +1,148 @@
+// test/helpers/merkleTree.js
+// SPDX-License-Identifier: MIT
+// from https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.0.1/test/helpers/merkleTree.js
+
+import { bufferToHex } from 'ethereumjs-util';
+import web3 from 'web3';
+
+export default class MerkleTree {
+  constructor(elements) {
+    // Filter empty strings and hash elements
+    this.elements = elements
+      .filter((el) => el)
+      .map((el) => Buffer.from(web3.utils.hexToBytes(web3.utils.sha3(el, { encoding: 'hex' }))));
+    // Sort elements
+    this.elements.sort(Buffer.compare);
+
+    // Deduplicate elements
+    this.elements = this.bufDedup(this.elements);
+
+    // Create layers
+    this.layers = this.getLayers(this.elements);
+  }
+
+  getLayers(elements) {
+    if (elements.length === 0) {
+      return [['']];
+    }
+
+    const layers = [];
+    layers.push(elements);
+
+    // Get next layer until we reach the root
+    while (layers[layers.length - 1].length > 1) {
+      layers.push(this.getNextLayer(layers[layers.length - 1]));
+    }
+
+    return layers;
+  }
+
+  getNextLayer(elements) {
+    return elements.reduce((layer, el, idx, arr) => {
+      if (idx % 2 === 0) {
+        // Hash the current element with its pair element
+        layer.push(this.combinedHash(el, arr[idx + 1]));
+      }
+
+      return layer;
+    }, []);
+  }
+
+  combinedHash(first, second) {
+    if (!first) {
+      return second;
+    }
+
+    if (!second) {
+      return first;
+    }
+
+    return Buffer.from(
+      web3.utils.hexToBytes(web3.utils.sha3(this.sortAndConcat(first, second), { encoding: 'hex' }))
+    );
+  }
+
+  getRoot() {
+    return this.layers[this.layers.length - 1][0];
+  }
+
+  getHexRoot() {
+    return bufferToHex(this.getRoot());
+  }
+
+  getProof(el) {
+    let idx = this.bufIndexOf(el, this.elements);
+
+    if (idx === -1) {
+      throw new Error('Element does not exist in Merkle tree');
+    }
+
+    console.log({ idx });
```

Comment:
There are some console logs in this file

Reply:
removed